run/test/api.ts

```diff
@@ -0,0 +1,127 @@
+import * as assert from 'assert';
+import {
+  Producer,
+  Subject,
+  pipe,
+  map,
+  filter,
+  uponStart,
+  makeSubject,
+  subscribe,
+} from '@cycle/callbags';
+
+import {
+  run,
+  applyApis,
+  IdGenerator,
+  Subscription,
+  Driver,
+  Plugin,
+  Api,
+} from '../src/index';
+
+type RandomSource = { id: number; value: number };
+
+class RandomApi implements Api<RandomSource> {
+  constructor(
+    public source: Producer<RandomSource>,
+    private sinkSubject: Subject<number>,
+    private gen: IdGenerator
+  ) {}
+
+  public get(): Producer<number> {
+    const id = this.gen();
+
+    return pipe(
+      this.source,
+      filter(x => x.id === id),
+      map(x => x.value),
+      uponStart(() => {
+        this.sinkSubject(2, id);
+      })
+    );
+  }
+}
+
+function randomApiFactory(
+  source: Producer<RandomSource>,
+  subject: Subject<number>,
+  gen: IdGenerator
+) {
+  return new RandomApi(source, subject, gen);
+}
+
+function makeRandomPlugin(n: number): Plugin<RandomSource, number> {
+  class RandomDriver implements Driver<RandomSource, number> {
+    private subject = makeSubject<RandomSource>();
+
+    provideSource() {
+      return this.subject;
+    }
+
+    consumeSink(sink: Producer<number>): Subscription {
+      return pipe(
+        sink,
+        subscribe(id => {
+          this.subject(2, { id, value: n });
```

Comment:
Question: I'm wondering, why `2`? Aren't we sending data, so shouldn't it be `1`?

Reply:
oh, yes you are right

Reply:
What I'm wondering is that why did the tests pass before if this was wrong with `2`? We would expect tests to fail.

(I unresolved, just so you can see the comment. Feel free to resolve again)

Reply:
ping again @jvanbruegge in case you didn't see this.

Reply:
Yes, I did indeed miss this. They passed most likely because i did not assert that `called === true` after `run`. I will add this just to be sure

Reply:
I've added the two assertions now and yes indeed that was the issue, the asserting code just was not run. Now the tests fail for `2` but not for `1`